samples/nextjs/src/pages/_edit.tsx

```diff
@@ -23,46 +18,19 @@ const SitecorePage = ({ layoutData }: SitecorePageProps): JSX.Element => {
   };
 
   const PageLayout = () => (
-    <SitecoreContext componentFactory={componentFactory} context={context}>
-      <Layout route={layoutData.sitecore.route} />
-    </SitecoreContext>
+    <ComponentPropsContext value={componentProps}>
+      <SitecoreContext componentFactory={componentFactory} context={context}>
+        <Layout route={layoutData.sitecore.route} />
+      </SitecoreContext>
+    </ComponentPropsContext>
   );
 
   return <PageLayout />;
 };
 
 SitecorePage.getInitialProps = async (context: NextPageContext) => {
-  const { req } = context;
-
-  // Grab Experience Editor data which has been stashed on the request
-  const data = (req as EditingRequest).editingData;
-
-  const props: SitecorePageProps = {
-    locale: data.language,
-    layoutData: data.layoutData,
-    dictionary: data.dictionary,
-    componentProps: {},
-  };
-
-  if (props?.layoutData?.sitecore.route) {
-    // Retrieve component props using side-effects defined on components level
-    props.componentProps = await componentPropsService.fetchInitialComponentProps({
-      layoutData: props.layoutData,
-      context,
-      componentModule,
-    });
-  }
-
-  //TODO: implement getInitialProps-compatible component props functionality
-  // if (props.layoutData) {
-  //   // Retrieve component props using side-effects defined on components level
-  //   props.componentProps = await componentPropsService.fetchInitialComponentProps({
-  //     layoutData: props.layoutData,
-  //     context,
-  //     componentModule,
-  //   });
-  // }
-
+  const propsFactory = new SitecorePagePropsFactory();
```

Comment:
@ambrauer For me it makes sense to create a `lib/page-props-factory.ts` and define it there, because this class doesn't contain arguments, and we don't need to create separate in every nextjs `page`

Reply:
@sc-illiakovalenko I agree, though eventually this may have constructor parms, it makes sense to export an instance and use that.